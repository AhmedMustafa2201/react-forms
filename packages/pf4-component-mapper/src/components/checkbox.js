import React from 'react';
import PropTypes from 'prop-types';
// import MultipleChoiceList from '../common/multiple-choice-list';
import { useFieldApi } from '@data-driven-forms/react-form-renderer';
import FormGroup from '../common/form-group';
import { Checkbox as Pf4Checkbox } from '@patternfly/react-core/dist/js/components/Checkbox/Checkbox';

const SingleCheckbox = (props) => {
  const { label, isRequired, helperText, meta, description, input, isReadOnly, isDisabled, id, ...rest } = useFieldApi(props);
  return (
    <FormGroup isRequired={isRequired} helperText={helperText} meta={meta} description={description} hideLabel id={id || input.name}>
      <Pf4Checkbox
        isChecked={input.checked}
        {...input}
        id={id || input.name}
        label={label}
        aria-label={rest.name}
        {...rest}
        isDisabled={isDisabled || isReadOnly}
      />
    </FormGroup>
  );
};

SingleCheckbox.propTypes = {
  label: PropTypes.string,
  isReadOnly: PropTypes.bool,
  isRequired: PropTypes.bool,
  helperText: PropTypes.string,
  meta: PropTypes.object.isRequired,
  description: PropTypes.string,
  input: PropTypes.shape({
    name: PropTypes.string.isRequired,
    checked: PropTypes.bool
  }).isRequired,
  isDisabled: PropTypes.bool,
  id: PropTypes.string
};

const Checkbox = ({ options, ...props }) => {
  /**
   * ignore multiple choice list for now
   * <MultipleChoiceList options={options} {...props} />
   */
  return <SingleCheckbox {...props} />;
};

Checkbox.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({ label: PropTypes.string, value: PropTypes.string }))
};

export default Checkbox;
