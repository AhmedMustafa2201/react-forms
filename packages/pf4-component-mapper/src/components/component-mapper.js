import componentTypes from '@data-driven-forms/react-form-renderer/dist/cjs/component-types';
import Tabs from './tabs';
import SubForm from './sub-form';
import Wizard from './wizard';
import Select from './select';
import { Select as RawSelect } from './select/select';
import FieldArray from './field-array';
import TextField from './text-field';
import TextArea from './text-area';
import Checkbox from './checkbox';
import Radio from './radio';
import DatePicker from './date-picker';
import TimePicker from './time-picker';
import Switch from './switch';
import PlainText from './plain-text';

const mapper = {
  [componentTypes.TEXT_FIELD]: TextField,
  [componentTypes.TEXTAREA_FIELD]: TextArea,
  [componentTypes.SELECT_COMPONENT]: Select,
  [componentTypes.CHECKBOX]: Checkbox,
  [componentTypes.SUB_FORM]: SubForm,
  [componentTypes.RADIO]: Radio,
  [componentTypes.TABS]: Tabs,
  [componentTypes.DATE_PICKER]: DatePicker,
  [componentTypes.TIME_PICKER]: TimePicker,
  [componentTypes.WIZARD]: Wizard,
  [componentTypes.SWITCH]: Switch,
  [componentTypes.PLAIN_TEXT]: PlainText,
  [componentTypes.FIELD_ARRAY]: FieldArray
};

export default mapper;

export const components = {
  TextField,
  TextArea,
  Checkbox,
  Radio,
  Select,
  DatePicker,
  TimePicker,
  Switch,
  PlainText
};

export const rawComponents = {
  RawSelect
};
