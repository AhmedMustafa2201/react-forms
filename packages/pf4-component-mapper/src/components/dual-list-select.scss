.ddorg__pf4-component-mapper__dual-list-select-toolbar {
    height: auto !important;
}

.ddorg__pf4-component-mapper__dual-list-select {
    height: 300px !important;
    background: none !important;
}

.ddorg__pf4-component-mapper__dual-list-select-option {
    height: 100% !important;
}
