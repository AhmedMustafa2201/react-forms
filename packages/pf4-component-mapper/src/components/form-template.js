import React from 'react';
import PropTypes from 'prop-types';

import formTemplate from '../../../common/src/form-template';

import { Text, TextVariants } from '@patternfly/react-core/dist/js/components/Text/Text';
import { TextContent } from '@patternfly/react-core/dist/js/components/Text/TextContent';
import { Form } from '@patternfly/react-core/dist/js/components/Form/Form';
import { ActionGroup } from '@patternfly/react-core/dist/js/components/Form/ActionGroup';
import { Button as PF4Button } from '@patternfly/react-core/dist/js/components/Button/Button';

import './form-template.scss';

export const Button = ({ label, bsStyle, children, disabled, ...props }) => (
  <PF4Button variant={bsStyle || 'secondary'} isDisabled={disabled} {...props}>
    {label}
    {children}
  </PF4Button>
);

Button.propTypes = {
  label: PropTypes.string.isRequired,
  bsStyle: PropTypes.string,
  disabled: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

export const ButtonGroup = ({ children, ...props }) => <ActionGroup {...props}>{children}</ActionGroup>;

ButtonGroup.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

export const Title = ({ children }) => (
  <TextContent>
    <Text component={TextVariants.h1}>{children}</Text>
  </TextContent>
);

Title.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

export const Description = ({ children }) => (
  <TextContent>
    <Text component={TextVariants.p}>{children}</Text>
  </TextContent>
);

Description.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

const PF4FormTemplate = (options) => formTemplate({ FormWrapper: Form, Button, ButtonGroup, Title, Description, ...options });

export default PF4FormTemplate;
