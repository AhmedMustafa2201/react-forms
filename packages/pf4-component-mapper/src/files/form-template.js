import React from 'react';
import PropTypes from 'prop-types';

import FormTemplate from '@data-driven-forms/common/src/form-template';

import { Button as PF4Button, ActionGroup, Form, TextContent, Text, TextVariants } from '@patternfly/react-core';

import './form-template.scss';

export const Button = ({ label, bsStyle, children, disabled, buttonType, ...props }) => (
  <PF4Button variant={bsStyle || 'secondary'} isDisabled={disabled} {...props}>
    {label}
    {children}
  </PF4Button>
);

Button.propTypes = {
  label: PropTypes.string.isRequired,
  bsStyle: PropTypes.string,
  disabled: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  buttonType: PropTypes.string
};

export const ButtonGroup = ({ children, ...props }) => <ActionGroup {...props}>{children}</ActionGroup>;

ButtonGroup.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

export const Title = ({ children }) => (
  <TextContent>
    <Text component={TextVariants.h1}>{children}</Text>
  </TextContent>
);

Title.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

export const Description = ({ children }) => (
  <TextContent>
    <Text component={TextVariants.p}>{children}</Text>
  </TextContent>
);

Description.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

const PF4FormTemplate = (props) => (
  <FormTemplate FormWrapper={Form} Button={Button} ButtonGroup={ButtonGroup} Title={Title} Description={Description} {...props} />
);

export default PF4FormTemplate;
