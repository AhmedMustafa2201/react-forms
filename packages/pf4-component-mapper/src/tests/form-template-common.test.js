import React from 'react';
import { Form } from '@data-driven-forms/react-form-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { formTemplate } from '../index';
import { Title, Description, Button } from '../components/form-template';
import RenderWithProvider from '../../../../__mocks__/with-provider';

describe('FormTemplate PF4 Common', () => {
  let initialProps;
  let ContextWrapper;
  let FormTemplate;
  let formOptions;

  beforeEach(() => {
    formOptions = { onSubmit: jest.fn(), onReset: jest.fn(), onCancel: jest.fn(), canReset: true, pristine: true };
    ContextWrapper = ({ children, ...props }) => (
      <RenderWithProvider value={{ formOptions }}>
        <Form onSubmit={jest.fn()}>{() => children}</Form>
      </RenderWithProvider>
    );
    initialProps = {
      formFields: <div>Formfields</div>,
      schema: {}
    };
  });

  it('should render title', () => {
    initialProps = {
      ...initialProps,
      schema: { title: 'I am title' }
    };

    FormTemplate = formTemplate({});

    const wrapper = mount(
      <ContextWrapper>
        <FormTemplate {...initialProps} />
      </ContextWrapper>
    );

    expect(wrapper.find(Title)).toHaveLength(1);
    expect(wrapper.find(Description)).toHaveLength(0);
    expect(wrapper.find(Button)).toHaveLength(2);
  });

  it('should hide buttons', () => {
    FormTemplate = formTemplate({ showFormControls: false });

    const wrapper = mount(
      <ContextWrapper>
        <FormTemplate {...initialProps} />
      </ContextWrapper>
    );

    expect(wrapper.find(Button)).toHaveLength(0);
  });

  it('should render description', () => {
    initialProps = {
      ...initialProps,
      schema: { description: 'I am description' }
    };

    FormTemplate = formTemplate({});

    const wrapper = mount(
      <ContextWrapper>
        <FormTemplate {...initialProps} />
      </ContextWrapper>
    );

    expect(wrapper.find(Title)).toHaveLength(0);
    expect(wrapper.find(Description)).toHaveLength(1);
  });

  it('should render custom formControls', () => {
    const FormButtons = () => <div>Form Controls</div>;

    FormTemplate = formTemplate({ FormButtons });

    const wrapper = mount(
      <ContextWrapper>
        <FormTemplate {...initialProps} />
      </ContextWrapper>
    );

    expect(wrapper.find(FormButtons)).toHaveLength(1);
  });

  it('should render all controls and with default labels', () => {
    FormTemplate = formTemplate({});

    const wrapper = mount(
      <ContextWrapper>
        <FormTemplate {...initialProps} />
      </ContextWrapper>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper
    .find('button')
    .first()
    .simulate('click');
    wrapper
    .find('button')
    .at(1)
    .simulate('click');
    wrapper
    .find('button')
    .last()
    .simulate('click');

    expect(formOptions.onSubmit).not.toHaveBeenCalled();
    expect(formOptions.onReset).not.toHaveBeenCalled();
    expect(formOptions.onCancel).toHaveBeenCalled();
  });

  it('should render only submit button', () => {
    FormTemplate = formTemplate({});

    const wrapper = mount(
      <ContextWrapper>
        <FormTemplate {...initialProps} canReset={false} onCancel={undefined} />
      </ContextWrapper>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render buttons in correct order', () => {
    FormTemplate = formTemplate({});

    const wrapper = mount(
      <ContextWrapper>
        <FormTemplate {...initialProps} canSubmit canReset buttonOrder={['cancel', 'submit', 'reset']} />
      </ContextWrapper>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should add missing buttons if not defined in button order', () => {
    FormTemplate = formTemplate({});

    const wrapper = mount(
      <ContextWrapper>
        <FormTemplate {...initialProps} canSubmit canReset buttonOrder={[]} />
      </ContextWrapper>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call cancel with form values', () => {
    FormTemplate = formTemplate({ canReset: true });

    const wrapper = mount(
      <ContextWrapper>
        <FormTemplate {...initialProps} />
      </ContextWrapper>
    );

    expect(wrapper.find(Button)).toHaveLength(3);

    const CANCEL_INDEX = 2;
    const FIELD_NAME = 'xxx';
    const FIELD_VALUE = 'yyy';

    const expectedValues = {
      [FIELD_NAME]: FIELD_VALUE
    };

    wrapper
    .find(Form)
    .instance()
    .form.change(FIELD_NAME, FIELD_VALUE);
    wrapper.update();

    wrapper
    .find('button')
    .at(CANCEL_INDEX)
    .simulate('click');

    expect(formOptions.onCancel).toHaveBeenCalledWith(expectedValues);
  });
});
