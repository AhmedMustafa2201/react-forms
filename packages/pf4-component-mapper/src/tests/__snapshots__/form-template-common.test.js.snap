// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTemplate PF4 Common should add missing buttons if not defined in button order 1`] = `
<ContextWrapper>
  <RenderWithProvider
    value={
      Object {
        "formOptions": Object {
          "canReset": true,
          "onCancel": [MockFunction],
          "onReset": [MockFunction],
          "onSubmit": [MockFunction],
          "pristine": true,
        },
      }
    }
  >
    <ReactFinalForm
      onSubmit={[Function]}
    >
      <ReactFinalForm
        onSubmit={[MockFunction]}
      >
        <Component
          buttonOrder={Array []}
          canReset={true}
          canSubmit={true}
          formFields={
            <div>
              Formfields
            </div>
          }
          schema={Object {}}
        >
          <Form>
            <form
              className="pf-c-form"
              noValidate={true}
            >
              <div>
                Formfields
              </div>
              <FormSpy>
                <FormControls
                  Button={[Function]}
                  ButtonGroup={[Function]}
                  FormSpy={[Function]}
                  buttonOrder={
                    Array [
                      "submit",
                      "reset",
                      "cancel",
                    ]
                  }
                  canReset={false}
                  canSubmit={false}
                  cancelLabel="Cancel"
                  formSpyProps={
                    Object {
                      "active": undefined,
                      "dirty": false,
                      "dirtyFields": Object {},
                      "dirtyFieldsSinceLastSubmit": Object {},
                      "dirtySinceLastSubmit": false,
                      "error": undefined,
                      "errors": Object {},
                      "form": Object {
                        "batch": [Function],
                        "blur": [Function],
                        "change": [Function],
                        "destroyOnUnregister": false,
                        "focus": [Function],
                        "getFieldState": [Function],
                        "getRegisteredFields": [Function],
                        "getState": [Function],
                        "initialize": [Function],
                        "isValidationPaused": [Function],
                        "mutators": Object {},
                        "pauseValidation": [Function],
                        "registerField": [Function],
                        "reset": [Function],
                        "resetFieldState": [Function],
                        "resumeValidation": [Function],
                        "setConfig": [Function],
                        "submit": [Function],
                        "subscribe": [Function],
                      },
                      "hasSubmitErrors": false,
                      "hasValidationErrors": false,
                      "initialValues": undefined,
                      "invalid": false,
                      "modified": Object {},
                      "pristine": true,
                      "submitError": undefined,
                      "submitErrors": undefined,
                      "submitFailed": false,
                      "submitSucceeded": false,
                      "submitting": false,
                      "touched": Object {},
                      "valid": true,
                      "validating": false,
                      "values": Object {},
                      "visited": Object {},
                    }
                  }
                  onCancel={[MockFunction]}
                  onReset={[MockFunction]}
                  resetLabel="Reset"
                  submitLabel="Submit"
                >
                  <ButtonGroup>
                    <ActionGroup>
                      <div
                        className="pf-c-form__group pf-m-action"
                      >
                        <div
                          className="pf-c-form__actions"
                        >
                          <Button
                            key="form-submit"
                            label="Submit"
                            type="submit"
                            variant="primary"
                          >
                            <Component
                              type="submit"
                              variant="primary"
                            >
                              <ComponentWithOuia
                                component={[Function]}
                                componentProps={
                                  Object {
                                    "children": Array [
                                      "Submit",
                                      undefined,
                                    ],
                                    "isDisabled": undefined,
                                    "type": "submit",
                                    "variant": "primary",
                                  }
                                }
                                consumerContext={null}
                              >
                                <Button
                                  ouiaContext={
                                    Object {
                                      "isOuia": false,
                                      "ouiaId": null,
                                    }
                                  }
                                  type="submit"
                                  variant="primary"
                                >
                                  <button
                                    aria-disabled={null}
                                    aria-label={null}
                                    className="pf-c-button pf-m-primary"
                                    disabled={false}
                                    tabIndex={null}
                                    type="submit"
                                  >
                                    Submit
                                  </button>
                                </Button>
                              </ComponentWithOuia>
                            </Component>
                          </Button>
                          <Button
                            key="form-cancel"
                            label="Cancel"
                            onClick={[Function]}
                            type="button"
                          >
                            <Component
                              onClick={[Function]}
                              type="button"
                              variant="secondary"
                            >
                              <ComponentWithOuia
                                component={[Function]}
                                componentProps={
                                  Object {
                                    "children": Array [
                                      "Cancel",
                                      undefined,
                                    ],
                                    "isDisabled": undefined,
                                    "onClick": [Function],
                                    "type": "button",
                                    "variant": "secondary",
                                  }
                                }
                                consumerContext={null}
                              >
                                <Button
                                  onClick={[Function]}
                                  ouiaContext={
                                    Object {
                                      "isOuia": false,
                                      "ouiaId": null,
                                    }
                                  }
                                  type="button"
                                  variant="secondary"
                                >
                                  <button
                                    aria-disabled={null}
                                    aria-label={null}
                                    className="pf-c-button pf-m-secondary"
                                    disabled={false}
                                    onClick={[Function]}
                                    tabIndex={null}
                                    type="button"
                                  >
                                    Cancel
                                  </button>
                                </Button>
                              </ComponentWithOuia>
                            </Component>
                          </Button>
                        </div>
                      </div>
                    </ActionGroup>
                  </ButtonGroup>
                </FormControls>
              </FormSpy>
            </form>
          </Form>
        </Component>
      </ReactFinalForm>
    </ReactFinalForm>
  </RenderWithProvider>
</ContextWrapper>
`;

exports[`FormTemplate PF4 Common should render all controls and with default labels 1`] = `
<ContextWrapper>
  <RenderWithProvider
    value={
      Object {
        "formOptions": Object {
          "canReset": true,
          "onCancel": [MockFunction],
          "onReset": [MockFunction],
          "onSubmit": [MockFunction],
          "pristine": true,
        },
      }
    }
  >
    <ReactFinalForm
      onSubmit={[Function]}
    >
      <ReactFinalForm
        onSubmit={[MockFunction]}
      >
        <Component
          formFields={
            <div>
              Formfields
            </div>
          }
          schema={Object {}}
        >
          <Form>
            <form
              className="pf-c-form"
              noValidate={true}
            >
              <div>
                Formfields
              </div>
              <FormSpy>
                <FormControls
                  Button={[Function]}
                  ButtonGroup={[Function]}
                  FormSpy={[Function]}
                  buttonOrder={
                    Array [
                      "submit",
                      "reset",
                      "cancel",
                    ]
                  }
                  canReset={false}
                  canSubmit={false}
                  cancelLabel="Cancel"
                  formSpyProps={
                    Object {
                      "active": undefined,
                      "dirty": false,
                      "dirtyFields": Object {},
                      "dirtyFieldsSinceLastSubmit": Object {},
                      "dirtySinceLastSubmit": false,
                      "error": undefined,
                      "errors": Object {},
                      "form": Object {
                        "batch": [Function],
                        "blur": [Function],
                        "change": [Function],
                        "destroyOnUnregister": false,
                        "focus": [Function],
                        "getFieldState": [Function],
                        "getRegisteredFields": [Function],
                        "getState": [Function],
                        "initialize": [Function],
                        "isValidationPaused": [Function],
                        "mutators": Object {},
                        "pauseValidation": [Function],
                        "registerField": [Function],
                        "reset": [Function],
                        "resetFieldState": [Function],
                        "resumeValidation": [Function],
                        "setConfig": [Function],
                        "submit": [Function],
                        "subscribe": [Function],
                      },
                      "hasSubmitErrors": false,
                      "hasValidationErrors": false,
                      "initialValues": undefined,
                      "invalid": false,
                      "modified": Object {},
                      "pristine": true,
                      "submitError": undefined,
                      "submitErrors": undefined,
                      "submitFailed": false,
                      "submitSucceeded": false,
                      "submitting": false,
                      "touched": Object {},
                      "valid": true,
                      "validating": false,
                      "values": Object {},
                      "visited": Object {},
                    }
                  }
                  onCancel={[MockFunction]}
                  onReset={[MockFunction]}
                  resetLabel="Reset"
                  submitLabel="Submit"
                >
                  <ButtonGroup>
                    <ActionGroup>
                      <div
                        className="pf-c-form__group pf-m-action"
                      >
                        <div
                          className="pf-c-form__actions"
                        >
                          <Button
                            key="form-submit"
                            label="Submit"
                            type="submit"
                            variant="primary"
                          >
                            <Component
                              type="submit"
                              variant="primary"
                            >
                              <ComponentWithOuia
                                component={[Function]}
                                componentProps={
                                  Object {
                                    "children": Array [
                                      "Submit",
                                      undefined,
                                    ],
                                    "isDisabled": undefined,
                                    "type": "submit",
                                    "variant": "primary",
                                  }
                                }
                                consumerContext={null}
                              >
                                <Button
                                  ouiaContext={
                                    Object {
                                      "isOuia": false,
                                      "ouiaId": null,
                                    }
                                  }
                                  type="submit"
                                  variant="primary"
                                >
                                  <button
                                    aria-disabled={null}
                                    aria-label={null}
                                    className="pf-c-button pf-m-primary"
                                    disabled={false}
                                    tabIndex={null}
                                    type="submit"
                                  >
                                    Submit
                                  </button>
                                </Button>
                              </ComponentWithOuia>
                            </Component>
                          </Button>
                          <Button
                            key="form-cancel"
                            label="Cancel"
                            onClick={[Function]}
                            type="button"
                          >
                            <Component
                              onClick={[Function]}
                              type="button"
                              variant="secondary"
                            >
                              <ComponentWithOuia
                                component={[Function]}
                                componentProps={
                                  Object {
                                    "children": Array [
                                      "Cancel",
                                      undefined,
                                    ],
                                    "isDisabled": undefined,
                                    "onClick": [Function],
                                    "type": "button",
                                    "variant": "secondary",
                                  }
                                }
                                consumerContext={null}
                              >
                                <Button
                                  onClick={[Function]}
                                  ouiaContext={
                                    Object {
                                      "isOuia": false,
                                      "ouiaId": null,
                                    }
                                  }
                                  type="button"
                                  variant="secondary"
                                >
                                  <button
                                    aria-disabled={null}
                                    aria-label={null}
                                    className="pf-c-button pf-m-secondary"
                                    disabled={false}
                                    onClick={[Function]}
                                    tabIndex={null}
                                    type="button"
                                  >
                                    Cancel
                                  </button>
                                </Button>
                              </ComponentWithOuia>
                            </Component>
                          </Button>
                        </div>
                      </div>
                    </ActionGroup>
                  </ButtonGroup>
                </FormControls>
              </FormSpy>
            </form>
          </Form>
        </Component>
      </ReactFinalForm>
    </ReactFinalForm>
  </RenderWithProvider>
</ContextWrapper>
`;

exports[`FormTemplate PF4 Common should render buttons in correct order 1`] = `
<ContextWrapper>
  <RenderWithProvider
    value={
      Object {
        "formOptions": Object {
          "canReset": true,
          "onCancel": [MockFunction],
          "onReset": [MockFunction],
          "onSubmit": [MockFunction],
          "pristine": true,
        },
      }
    }
  >
    <ReactFinalForm
      onSubmit={[Function]}
    >
      <ReactFinalForm
        onSubmit={[MockFunction]}
      >
        <Component
          buttonOrder={
            Array [
              "cancel",
              "submit",
              "reset",
            ]
          }
          canReset={true}
          canSubmit={true}
          formFields={
            <div>
              Formfields
            </div>
          }
          schema={Object {}}
        >
          <Form>
            <form
              className="pf-c-form"
              noValidate={true}
            >
              <div>
                Formfields
              </div>
              <FormSpy>
                <FormControls
                  Button={[Function]}
                  ButtonGroup={[Function]}
                  FormSpy={[Function]}
                  buttonOrder={
                    Array [
                      "submit",
                      "reset",
                      "cancel",
                    ]
                  }
                  canReset={false}
                  canSubmit={false}
                  cancelLabel="Cancel"
                  formSpyProps={
                    Object {
                      "active": undefined,
                      "dirty": false,
                      "dirtyFields": Object {},
                      "dirtyFieldsSinceLastSubmit": Object {},
                      "dirtySinceLastSubmit": false,
                      "error": undefined,
                      "errors": Object {},
                      "form": Object {
                        "batch": [Function],
                        "blur": [Function],
                        "change": [Function],
                        "destroyOnUnregister": false,
                        "focus": [Function],
                        "getFieldState": [Function],
                        "getRegisteredFields": [Function],
                        "getState": [Function],
                        "initialize": [Function],
                        "isValidationPaused": [Function],
                        "mutators": Object {},
                        "pauseValidation": [Function],
                        "registerField": [Function],
                        "reset": [Function],
                        "resetFieldState": [Function],
                        "resumeValidation": [Function],
                        "setConfig": [Function],
                        "submit": [Function],
                        "subscribe": [Function],
                      },
                      "hasSubmitErrors": false,
                      "hasValidationErrors": false,
                      "initialValues": undefined,
                      "invalid": false,
                      "modified": Object {},
                      "pristine": true,
                      "submitError": undefined,
                      "submitErrors": undefined,
                      "submitFailed": false,
                      "submitSucceeded": false,
                      "submitting": false,
                      "touched": Object {},
                      "valid": true,
                      "validating": false,
                      "values": Object {},
                      "visited": Object {},
                    }
                  }
                  onCancel={[MockFunction]}
                  onReset={[MockFunction]}
                  resetLabel="Reset"
                  submitLabel="Submit"
                >
                  <ButtonGroup>
                    <ActionGroup>
                      <div
                        className="pf-c-form__group pf-m-action"
                      >
                        <div
                          className="pf-c-form__actions"
                        >
                          <Button
                            key="form-submit"
                            label="Submit"
                            type="submit"
                            variant="primary"
                          >
                            <Component
                              type="submit"
                              variant="primary"
                            >
                              <ComponentWithOuia
                                component={[Function]}
                                componentProps={
                                  Object {
                                    "children": Array [
                                      "Submit",
                                      undefined,
                                    ],
                                    "isDisabled": undefined,
                                    "type": "submit",
                                    "variant": "primary",
                                  }
                                }
                                consumerContext={null}
                              >
                                <Button
                                  ouiaContext={
                                    Object {
                                      "isOuia": false,
                                      "ouiaId": null,
                                    }
                                  }
                                  type="submit"
                                  variant="primary"
                                >
                                  <button
                                    aria-disabled={null}
                                    aria-label={null}
                                    className="pf-c-button pf-m-primary"
                                    disabled={false}
                                    tabIndex={null}
                                    type="submit"
                                  >
                                    Submit
                                  </button>
                                </Button>
                              </ComponentWithOuia>
                            </Component>
                          </Button>
                          <Button
                            key="form-cancel"
                            label="Cancel"
                            onClick={[Function]}
                            type="button"
                          >
                            <Component
                              onClick={[Function]}
                              type="button"
                              variant="secondary"
                            >
                              <ComponentWithOuia
                                component={[Function]}
                                componentProps={
                                  Object {
                                    "children": Array [
                                      "Cancel",
                                      undefined,
                                    ],
                                    "isDisabled": undefined,
                                    "onClick": [Function],
                                    "type": "button",
                                    "variant": "secondary",
                                  }
                                }
                                consumerContext={null}
                              >
                                <Button
                                  onClick={[Function]}
                                  ouiaContext={
                                    Object {
                                      "isOuia": false,
                                      "ouiaId": null,
                                    }
                                  }
                                  type="button"
                                  variant="secondary"
                                >
                                  <button
                                    aria-disabled={null}
                                    aria-label={null}
                                    className="pf-c-button pf-m-secondary"
                                    disabled={false}
                                    onClick={[Function]}
                                    tabIndex={null}
                                    type="button"
                                  >
                                    Cancel
                                  </button>
                                </Button>
                              </ComponentWithOuia>
                            </Component>
                          </Button>
                        </div>
                      </div>
                    </ActionGroup>
                  </ButtonGroup>
                </FormControls>
              </FormSpy>
            </form>
          </Form>
        </Component>
      </ReactFinalForm>
    </ReactFinalForm>
  </RenderWithProvider>
</ContextWrapper>
`;

exports[`FormTemplate PF4 Common should render only submit button 1`] = `
<ContextWrapper>
  <RenderWithProvider
    value={
      Object {
        "formOptions": Object {
          "canReset": true,
          "onCancel": [MockFunction],
          "onReset": [MockFunction],
          "onSubmit": [MockFunction],
          "pristine": true,
        },
      }
    }
  >
    <ReactFinalForm
      onSubmit={[Function]}
    >
      <ReactFinalForm
        onSubmit={[MockFunction]}
      >
        <Component
          canReset={false}
          formFields={
            <div>
              Formfields
            </div>
          }
          schema={Object {}}
        >
          <Form>
            <form
              className="pf-c-form"
              noValidate={true}
            >
              <div>
                Formfields
              </div>
              <FormSpy>
                <FormControls
                  Button={[Function]}
                  ButtonGroup={[Function]}
                  FormSpy={[Function]}
                  buttonOrder={
                    Array [
                      "submit",
                      "reset",
                      "cancel",
                    ]
                  }
                  canReset={false}
                  canSubmit={false}
                  cancelLabel="Cancel"
                  formSpyProps={
                    Object {
                      "active": undefined,
                      "dirty": false,
                      "dirtyFields": Object {},
                      "dirtyFieldsSinceLastSubmit": Object {},
                      "dirtySinceLastSubmit": false,
                      "error": undefined,
                      "errors": Object {},
                      "form": Object {
                        "batch": [Function],
                        "blur": [Function],
                        "change": [Function],
                        "destroyOnUnregister": false,
                        "focus": [Function],
                        "getFieldState": [Function],
                        "getRegisteredFields": [Function],
                        "getState": [Function],
                        "initialize": [Function],
                        "isValidationPaused": [Function],
                        "mutators": Object {},
                        "pauseValidation": [Function],
                        "registerField": [Function],
                        "reset": [Function],
                        "resetFieldState": [Function],
                        "resumeValidation": [Function],
                        "setConfig": [Function],
                        "submit": [Function],
                        "subscribe": [Function],
                      },
                      "hasSubmitErrors": false,
                      "hasValidationErrors": false,
                      "initialValues": undefined,
                      "invalid": false,
                      "modified": Object {},
                      "pristine": true,
                      "submitError": undefined,
                      "submitErrors": undefined,
                      "submitFailed": false,
                      "submitSucceeded": false,
                      "submitting": false,
                      "touched": Object {},
                      "valid": true,
                      "validating": false,
                      "values": Object {},
                      "visited": Object {},
                    }
                  }
                  onCancel={[MockFunction]}
                  onReset={[MockFunction]}
                  resetLabel="Reset"
                  submitLabel="Submit"
                >
                  <ButtonGroup>
                    <ActionGroup>
                      <div
                        className="pf-c-form__group pf-m-action"
                      >
                        <div
                          className="pf-c-form__actions"
                        >
                          <Button
                            key="form-submit"
                            label="Submit"
                            type="submit"
                            variant="primary"
                          >
                            <Component
                              type="submit"
                              variant="primary"
                            >
                              <ComponentWithOuia
                                component={[Function]}
                                componentProps={
                                  Object {
                                    "children": Array [
                                      "Submit",
                                      undefined,
                                    ],
                                    "isDisabled": undefined,
                                    "type": "submit",
                                    "variant": "primary",
                                  }
                                }
                                consumerContext={null}
                              >
                                <Button
                                  ouiaContext={
                                    Object {
                                      "isOuia": false,
                                      "ouiaId": null,
                                    }
                                  }
                                  type="submit"
                                  variant="primary"
                                >
                                  <button
                                    aria-disabled={null}
                                    aria-label={null}
                                    className="pf-c-button pf-m-primary"
                                    disabled={false}
                                    tabIndex={null}
                                    type="submit"
                                  >
                                    Submit
                                  </button>
                                </Button>
                              </ComponentWithOuia>
                            </Component>
                          </Button>
                          <Button
                            key="form-cancel"
                            label="Cancel"
                            onClick={[Function]}
                            type="button"
                          >
                            <Component
                              onClick={[Function]}
                              type="button"
                              variant="secondary"
                            >
                              <ComponentWithOuia
                                component={[Function]}
                                componentProps={
                                  Object {
                                    "children": Array [
                                      "Cancel",
                                      undefined,
                                    ],
                                    "isDisabled": undefined,
                                    "onClick": [Function],
                                    "type": "button",
                                    "variant": "secondary",
                                  }
                                }
                                consumerContext={null}
                              >
                                <Button
                                  onClick={[Function]}
                                  ouiaContext={
                                    Object {
                                      "isOuia": false,
                                      "ouiaId": null,
                                    }
                                  }
                                  type="button"
                                  variant="secondary"
                                >
                                  <button
                                    aria-disabled={null}
                                    aria-label={null}
                                    className="pf-c-button pf-m-secondary"
                                    disabled={false}
                                    onClick={[Function]}
                                    tabIndex={null}
                                    type="button"
                                  >
                                    Cancel
                                  </button>
                                </Button>
                              </ComponentWithOuia>
                            </Component>
                          </Button>
                        </div>
                      </div>
                    </ActionGroup>
                  </ButtonGroup>
                </FormControls>
              </FormSpy>
            </form>
          </Form>
        </Component>
      </ReactFinalForm>
    </ReactFinalForm>
  </RenderWithProvider>
</ContextWrapper>
`;
